src/task.rs

```diff
@@ -0,0 +1,421 @@
+//! Tasks are lightweight, isolated, pre-emptively scheduled JavaScript
+//! execution threads.
+//!
+//! One of the most foundational properties of Starling is that no individual
+//! misbehaving JavaScript task can bring down the whole system.
+//!
+//! Tasks are **pre-emptively scheduled**. If one task goes into an infinite
+//! loop, or generally becomes CPU bound for a sustained period of time, other
+//! tasks in the system are not starved of resources, and system latency does
+//! not fall off a cliff.
+//!
+//! Tasks are **isolated** from each other and share no memory. They communicate
+//! by **message passing** over a readable and writable streams. This means that
+//! if one task's local state is corrupted, it doesn't immediately infect all
+//! other tasks' states that are communicating with it. The corrupted task can
+//! be killed by its supervisor and then respawned with the last known good
+//! state or with a clean slate.
+//!
+//! Tasks are **lightweight**. In order to facilitate let-it-fail error handling
+//! coupled with supervision hierarchies, idle tasks have little memory
+//! overhead, and near no time overhead.
```

Comment:
I realize that much of this introductory comment is aspirational right now, but this part seems somewhat farther out than the rest in that it depends on SpiderMonkey stuff. Specifically, at the very least [bug 1323066](https://bugzilla.mozilla.org/show_bug.cgi?id=1323066) needs to be done. Perhaps extend the comment to be explicit about that?

Reply:
Sure thing.